feature/experimental/gribi/otg_tests/encap_decap_scale/README.md

```diff
@@ -0,0 +1,312 @@
+    *   Add 4 VRFs for encapsulations: `ENCAP_TE_VRF_A`, `ENCAP_TE_VRF_B`, `ENCAP_TE_VRF_C` and `ENCAP_TE_VRF_D`.
+    *   Inject 5000 IPv4Entry-ies and 5000 IPv6Entry-ies to each of the 4 encap VRFs.
+    *   The entries in the encap VRFs should point to NextHopGroups in the `DEFAULT` VRF. Inject 200 such NextHopGroups in the DEFAULT VRF.
+    *   Each NextHopGroup should have 8 NextHops where each NextHop points to a tunnel in the `TE_VRF_111`. In addition, the weights specified in the NextHopGroup should be co-prime and the sum of the weights should be 16.
+    *   Add 1 VRF for decapsulation, `DECAP_TE_VRF`.
+    *   Inject `48` entries in the DECAP_TE_VRF where the entries have a mix of prefix lengths /22, /24, /26, and /28.
+2.  Send the following packets to DUT-1
+
+    ```
+    * inner_src: `ipv4_inner_src`
+    * inner_dst: `ipv4_inner_encap_match`
+    * dscp: `dscp_encap_a`
+    * outter_src: `ipv4_outter_src_222`
+    * outter_dst: `ipv4_outter_decap_match`
+    * dscp: `dscp_encap_a`
+    * proto: `4`
+
+    * inner_src: `ipv6_inner_src`
+    * inner_dst: `ipv6_inner_encap_match`
+    * dscp: `dscp_encap_a`
+    * outter_src: `ipv4_outter_src_111`
+    * outter_dst: `ipv4_outter_decap_match`
+    * dscp: `dscp_encap_a`
+    * proto: `41`
+
+    * inner_src: `ipv4_inner_src`
+    * inner_dst: `ipv4_inner_encap_match`
+    * dscp: `dscp_encap_b`
+    * outter_src: `ipv4_outter_src_222`
+    * outter_dst: `ipv4_outter_decap_match`
+    * dscp: `dscp_encap_b`
+    * proto: `4`
+
+    * inner_src: `ipv6_inner_src`
+    * inner_dst: `ipv6_inner_encap_match`
+    * dscp: `dscp_encap_b`
+    * outter_src: `ipv4_outter_src_111`
+    * outter_dst: `ipv4_outter_decap_match`
+    * dscp: `dscp_encap_b`
+    * proto: `41`
+
+    * inner_src: `ipv4_inner_src`
+    * inner_dst: `ipv4_inner_encap_match`
+    * dscp: `dscp_encap_c`
+    * outter_src: `ipv4_outter_src_222`
+    * outter_dst: `ipv4_outter_decap_match`
+    * dscp: `dscp_encap_c`
+    * proto: `4`
+
+    * inner_src: `ipv6_inner_src`
+    * inner_dst: `ipv6_inner_encap_match`
+    * dscp: `dscp_encap_c`
+    * outter_src: `ipv4_outter_src_111`
+    * outter_dst: `ipv4_outter_decap_match`
+    * dscp: `dscp_encap_c`
+    * proto: `41`
+
+    * inner_src: `ipv4_inner_src`
+    * inner_dst: `ipv4_inner_encap_match`
+    * dscp: `dscp_encap_d`
+    * outter_src: `ipv4_outter_src_222`
+    * outter_dst: `ipv4_outter_decap_match`
+    * dscp: `dscp_encap_d`
+    * proto: `4`
+
+    * inner_src: `ipv6_inner_src`
+    * inner_dst: `ipv6_inner_encap_match`
+    * dscp: `dscp_encap_d`
+    * outter_src: `ipv4_outter_src_111`
+    * outter_dst: `ipv4_outter_decap_match`
+    * dscp: `dscp_encap_d`
+    * proto: `41`    
+    ```
+
+3.  Send traffic to DUT-1, covering all the installed v4 and v6 entries in the decap and encap VRFs. Validate that all traffic are all decapped per the DECAP VRFs and then encapsulated per the ENCAP VRFs and received as encapsulated packet by ATE.
```

Comment:
I _think_ this means that the `ipv*_inner_src` and `ipv*_inner_encap_match` variables have to come from these prefixes, but it would be helpful if the text was more explicit about this.

Reply:
This README is given by google.